airbyte-integrations/connectors/source-close-com/source_close_com/source.py

```diff
@@ -0,0 +1,479 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+from abc import ABC, abstractmethod
+from base64 import b64encode
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+from urllib.parse import parse_qsl, urlparse
+
+import requests
+from airbyte_cdk.logger import AirbyteLogger
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+
+
+class CloseComStream(HttpStream, ABC):
+    url_base: str = "https://api.close.com/api/v1/"
+    primary_key: str = "id"
+    limit: int = 100
+
+    def __init__(self, **kwargs: Mapping[str, Any]):
+        super().__init__(authenticator=kwargs["authenticator"])
+        self.config: Mapping[str, Any] = kwargs
+        self.start_date: str = kwargs["start_date"]
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        decoded_response = response.json()
+        if bool(decoded_response.get("has_more", None)) and decoded_response.get("data", []):
+            parsed = dict(parse_qsl(urlparse(response.url).query))
+            # close.com has default skip param - 0
+            skip = parsed.get("_skip", 0)
+            limit = parsed.get("_limit", self.limit)
+            return {"_skip": int(skip) + int(limit)}
+        return None
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+
+        params = {"_limit": self.limit}
+
+        # Handle pagination by inserting the next page's token in the request parameters
+        if next_page_token:
+            params.update(next_page_token)
+
+        return params
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        yield from response.json()["data"]
+
+    def backoff_time(self, response: requests.Response) -> Optional[float]:
+        """This method is called if we run into the rate limit.
+        Close Com puts the retry time in the `rate_reset` response body so
+        we return that value. If the response is anything other than a 429 (e.g: 5XX)
+        fall back on default retry behavior.
+        Rate Limits Docs: https://developer.close.com/#ratelimits"""
+
+        return response.json().get("error", {}).get("rate_reset", 0)
+
+
+class IncrementalCloseComStream(CloseComStream, ABC):
+    @property
+    @abstractmethod
+    def cursor_field(self) -> str:
```

Comment:
I can see you have cursor_field in every stream declared directly at the class level, therefore there is no need to override this property, remove it.